lib/logmatcher.c

```diff
@@ -675,7 +675,7 @@ log_matcher_pcre_re_replace(LogMatcher *s, LogMessage *msg, gint value_handle, c
   if (new_value)
     {
       /* append the last literal */
-      g_string_append_len(new_value, &value[last_offset], value_len - last_offset);
+      g_string_append_len(new_value, &result.source_value[last_offset], value_len - last_offset);
```

Comment:
I really like this fix! 

One minor thing: can you replace `value_len` with `result.source_value_len` in this function, please so it will be consistent?